pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -422,6 +429,43 @@ func (tkn *SQLTokenizer) scanFormatParameter(prefix rune) (TokenKind, []byte) {
 	return Variable, tkn.bytes()
 }
 
+// scanDollarQuotedString scans a Postgres dollar-quoted string constant.
+// See: https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING
+func (tkn *SQLTokenizer) scanDollarQuotedString() (TokenKind, []byte) {
+	kind, tag := tkn.scanString('$', String)
+	if kind == LexError {
+		return kind, tkn.bytes()
+	}
+	var (
+		got int
+		buf bytes.Buffer
+	)
+	delim := string(tag)
+	if delim != "$$" {
+		delim = "$" + delim + "$"
+	}
+	for {
+		ch := tkn.lastChar
+		tkn.advance()
+		if ch == EndChar {
+			tkn.setErr("unexpected EOF in dollar-quoted string")
+			return LexError, buf.Bytes()
+		}
+		if byte(ch) == delim[got] {
+			got++
+			if got == len(delim) {
+				break
+			}
+			continue
+		}
+		if got > 0 {
```

Comment:
shouldn't you reset the buf here?

Reply:
actually, what is this function supposed to return? The interior of the quoted string right?

If we get $one$$onetwo$$onetwo$$one$, should it return `$onetwo$$onetwo$` ?